### ISSUE TITLE
Tuya Cloud Configured but Add a New Device is Blank

### ISSUE THREAD

**Issue by kintaroju** [rospogrigio/localtuya#1252]
Hi,

I've had my Tuya IoT account/project setup with Tuya official HA integration setup for over a year, and I decided to make the jump to Local Tuya. The issue I am facing is that the "add a new device" comes up with only the ... to manually add a device. 

![image](https://user-images.githubusercontent.com/29717036/213323443-ba2ebb1f-435d-47a1-b178-9a1aef71a145.png)

Am i missing something to have the devices show up in local tuya, as I can see them in HA using the official integration in HA and also in the IoT Tuya website?

Cheers

**Comment by allan-null** [comment]
The same is happening to me as a new user of this project. I am migrating from the official Tuya integration as well.

**Comment by kintaroju** [comment]
Hi,

On a separate side note, I did manually add the required entries, IP, ip/key and it actually connected to the device without issue and reports to HA.

On another note I was watching a youtube video about doing local tuya, and one person had a point about vlans, and my Tuya devices are on a separate vlan, so I am not sure if the auto discovery will only work for devices on the same vlan or can work across vlans? 

If someone can clarify the vlan question, that would be cool, thanks

**Comment by bucky2076** [comment]
i'm experiencing the same problem. iot devices are located in a different vlan... but firewall between the server and the iot devices are open (though not the other way around)

**Comment by paul-g-ionescu** [comment]
I have the same issue, and I don't have any vlans, just one single lan with IoT devices and HA box in it.

**Comment by kintaroju** [comment]
> I have the same issue, and I don't have any vlans, just one single lan with IoT devices and HA box in it.

Hi Paul,

besides you not seeing it, are you at least able to manually configure it and add items to local tuya and have it interact with your devices?

**Comment by paul-g-ionescu** [comment]
> > I have the same issue, and I don't have any vlans, just one single lan with IoT devices and HA box in it.
> 
> Hi Paul,
> 
> besides you not seeing it, are you at least able to manually configure it and add items to local tuya and have it interact with your devices?

No, I cannot configure it manually either. I get an error that _**"Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists."**_
I tried several times with all protocol versions.
This is a LIDL Zigbee Gateway that has connected few zigbee lights and movement sensors.

The official HA Tuya integration works ok with these devices, but is annoying to have to use the cloud API for using them, and to renew the demo account every months or so.

**Comment by kintaroju** [comment]
Hi Paul,

I believe local tuya doesn't support zigbee to my knowledge. 

Have you tried to manually provision it as a zigbee device in HA?

https://community.home-assistant.io/t/local-control-tuya-zigbee-devices-without-internet-localtuya/434692/5

**Comment by paul-g-ionescu** [comment]
> 

Well, it's a zigbee gateway, so it has Wired Ethernet on one side and a Zigbee radio interface to connect to zigbee devices.
I was assuming that I can connect to this gateway via IP locally and then access the zigbee devices managed through it.

**Comment by haywiremk** [comment]
I have the exact same issue. Have to manually config and have my devices on separate vlan and do allow internet access (there was a Note 2) about this.  Oddly HA discovers and recommends the Tuya integration.  So seems maybe something with this Local Tuya Integration.
